libres/lib/enkf/enkf_main.cpp

```diff
@@ -743,6 +741,320 @@ static FILE *enkf_main_log_step_list(const char *log_path,
+
+    std::unordered_map<
+        std::string,
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>>
+        parameters;
+    for (auto &[dataset_name, dataset] :
+         local_ministep_get_datasets(ministep)) {
+
+        if (!(local_dataset_get_size(dataset)))
+            continue;
+
+        const auto &scaled_keys = local_dataset_scaled_keys(dataset);
+
+        if (scaled_keys.size() == 0)
+            continue;
+
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>
+            row_scaling_list;
+        for (int ikw = 0; ikw < scaled_keys.size(); ikw++) {
+            const auto &key = scaled_keys[ikw];
+            const active_list_type *active_list =
+                local_dataset_get_node_active_list(dataset, key.c_str());
+            const int matrix_rows = matrix_get_rows(A);
+            const auto *config_node =
+                ensemble_config_get_node(ensemble_config, key.c_str());
+            const int node_size =
+                enkf_config_node_get_data_size(config_node, last_step);
+            if (matrix_get_rows(A) < node_size)
+                matrix_resize(A, node_size, active_ens_size, false);
+
+            for (int iens = 0; iens < int_vector_size(iens_active_index);
+                 iens++) {
+                int column = int_vector_iget(iens_active_index, iens);
+                if (column >= 0) {
+                    serialize_node(target_fs, config_node, iens, last_step, 0,
+                                   column, active_list, A);
+                }
+            }
+            const row_scaling_type *row_scaling =
+                local_dataset_get_row_scaling(dataset, key.c_str());
+            matrix_shrink_header(A, row_scaling->size(), matrix_get_columns(A));
+            row_scaling_list.push_back(
+                std::pair<matrix_type *, const row_scaling_type *>(
+                    matrix_alloc_copy(A), row_scaling));
+        }
+        parameters[std::string(dataset_name)] = row_scaling_list;
+    }
+    matrix_free(A);
+    return parameters;
+}
+
+static void enkf_main_save_row_scaling_parameters(
+    enkf_fs_type *target_fs, ensemble_config_type *ensemble_config,
+    int_vector_type *iens_active_index, int last_step,
+    const local_ministep_type *ministep,
+    std::unordered_map<
+        std::string,
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>>
+        parameters) {
+
+    for (auto &[dataset_name, dataset] :
+         local_ministep_get_datasets(ministep)) {
+
+        if (!(local_dataset_get_size(dataset)))
+            continue;
+
+        const auto &scaled_keys = local_dataset_scaled_keys(dataset);
+
+        if (scaled_keys.size() == 0)
+            continue;
+
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>
+            row_scaling_list = parameters[dataset_name];
+        for (int ikw = 0; ikw < scaled_keys.size(); ikw++) {
```

Comment:
Maybe use a range-for here? (https://en.cppreference.com/w/cpp/language/range-for)
```
for(auto key : row_scaling_list) {
  
}
```